src/java/edu/slu/action/ObjectAction.java

```diff
@@ -1412,24 +1453,42 @@ public void saveNewObject() throws IOException, ServletException, Exception{
                     // cubap: if we want, we can just copy the Slug to @id, warning
                     // if there was some mismatch, since versions are fine with that.
                 }
-                String newObjectID = mongoDBService.save(Constant.COLLECTION_ANNOTATION, dbo);
+                /*Table table = dynamoDB.getTable(tableName);
+                Item item = new Item().withPrimaryKey("ID", "friday-insert")
+                                      .withNumber("Misc_Field", 21004)
+                                      .withNumber("Random_Char", 2500);
+	        table.putItem(item);*/
+                    UUID uniqueKey = UUID.randomUUID();
+                    String primaryKeyId = uniqueKey.toString();
+                    Table table = dynamoDB.getTable(tableName);
+                    String newPrimaryKeyId = Constant.RERUM_ID_PREFIX+primaryKeyId;
+                Item item = new Item().withPrimaryKey("id", newPrimaryKeyId)
+                                      .withJSON(Constant.COLLECTION_ANNOTATION, received.toString());
+	        table.putItem(item);
```

Comment:
I [looked this one up](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/AmazonDynamoDB.html#putItem-com.amazonaws.services.dynamodbv2.model.PutItemRequest-):

> To prevent a new item from replacing an existing item, use a conditional expression that contains the attribute_not_exists function with the name of the attribute being used as the partition key for the table. Since every record must contain that attribute, the attribute_not_exists function will only succeed if no matching item exists.

In "save a new item" we definitely need to enforce that it is indeed new with appropriate handling if it is not.